DataDog/datadog-agent
# [system-probe] Add telemetry for dropped DNS stats

### What does this PR do?

In the last release, we added #7665 to limit the number of DNS stats we process.
In hosts with high DNS query rates, this may cause large number of DNS stats to be dropped.

To get visibility into impact on customers, we need to add telemetry for dropped stats.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

- Edit `etc/datadog-agent/system-probe.yaml` and set `max_dns_stats` to a really small number:
```yaml
system_probe_config:
  ....
  max_dns_stats: 2
```
- Restart `system-probe`.
- `curl` a few websites to generate DNS queries.
- Run `curl -s --unix-socket /opt/datadog-agent/run/sysprobe.sock http://unix/connections?client_id=1 | jq .connTelemetry`
- Should see something like the following in the output:
```
{
   ....
  "dnsStatsDropped": "2"
}
```
